django_cloud_deploy/cli/prompt_manager.py

```diff
@@ -0,0 +1,875 @@
+
+    return answer
+
+
+def _multiple_choice_validate(s: str, default: Optional[str], len_options: int):
+    """Validates the option chosen is valid.
+
+    Args:
+        s: Value to validate.
+        default: Default value if user provides no value. (Presses enter)
+        len_options: Amount of possible options for the user.
+
+    Raises:
+        ValueError: If the answer is not valid.
+    """
+    if default is not None and s == '':
+        return
+
+    if not str.isnumeric(s):
+        raise ValueError('Please enter a numeric value')
+
+    if 1 <= int(s) <= (len_options + 1):
+        return
+    else:
+        raise ValueError('Value is not in range')
+
+
+def _binary_prompt(question: str, console: io.IO,
+                   default: Optional[str] = None):
+    """Used to prompt user to choose from a yes or no question.
+
+    Args:
+        question: Question shown to the user on the console.
+        console: Object to use for user I/O.
+        default: Default value if user provides no value. (Presses enter)
+    """
+
+    while True:
+        try:
+            answer = console.ask(question)
+            if default and answer is '':
+                answer = default
+                _binary_validate(answer)
+            break
+        except ValueError as e:
+            console.error(e)
+
+    return answer
+
+
+def _binary_validate(s: str):
+    """Ensures value is yes or no.
+
+    Args:
+        s: Value to validate.
+    """
+    if s.lower() not in ['y', 'n']:
+        raise ValueError('Please respond using "y" or "n"')
+
+    return
+
+
+def _password_prompt(question: str, console: io.IO) -> str:
+    """Used to prompt user to choose a password field.
+
+    Args:
+        console: Object to use for user I/O.
+        question: Question shown to the user on the console.
+    """
+    console.tell(question)
+    while True:
+        password1 = console.getpass('Password: ')
+        try:
+            _password_validate(password1)
+        except ValueError as e:
+            console.error(e)
+            continue
+        password2 = console.getpass('Password (again): ')
+        if password1 != password2:
+            console.error('Passwords do not match, please try again')
+            continue
+        return password1
+
+
+def _password_validate(s):
+    """Validates that a string is a valid password.
+
+    Args:
+        s: The string to validate.
+
+    Raises:
+        ValueError: if the input string is not valid.
+    """
+    if len(s) < 5:
+        raise ValueError('Passwords must be at least 6 characters long')
+    allowed_characters = frozenset(string.ascii_letters + string.digits +
+                                   string.punctuation)
+    if frozenset(s).issuperset(allowed_characters):
+        raise ValueError('Invalid character in password: '
+                         'use letters, numbers and punctuation')
+
+    return
+
+
+class TemplatePrompt(object):
```

Comment:
Should there be a `Prompt` that is an abc that this (which is also an abc) is a subclass of?